### PR TITLE
feat: support providing a repository name to `createClient()` in place of an API endpoint

### PR DESCRIPTION
**Pull request by angeloashmore**
## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

**IMPORTANT**: This PR depends on #222. DO NOT MERGE until #222 is merged.

This PR adds support for providing a repository name to `createClient()` in place of an API endpoint.

```ts
import { createClient } from "@prismicio/client";

const client = prismic.createClient("repo-name");
```

Passing an API endpoint is still supported:

```ts
import { createClient } from "@prismicio/client";

const endpoint = prismic.getRepositoryEndpoint("repo-name");
const client = prismic.createClient(endpoint);
```

If the first argument for `createClient()` is not an endpoint, it will build one using `getRepositoryEndpoint()` automatically.

This takes advantage of the `isRepositoryEndpoint()` helper introduced in #222.

Related: https://github.com/prismicio/prismic-helpers/issues/42

## Checklist:

- [x] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

🦄

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -319,26 +331,31 @@ export class Client {
...
-			/\.prismic\.io\/(?!api\/v2\/?)/.test(endpoint)
+			/\.prismic\.io\/(?!api\/v2\/?)/.test(repositoryNameOrEndpoint)
 		) {
 			throw new PrismicError(
-				"@prismicio/client only supports Prismic Rest API V2. Please use the getEndpoint helper to generate a valid Rest API V2 endpoint URL.",
+				"@prismicio/client only supports Prismic Rest API V2. Please use the getRepositoryEndpoint helper to generate a valid Rest API V2 endpoint URL.",
 				undefined,
 				undefined,
 			);
 		}
```
Comment (lihbr) [inline src/client.ts:351]: While it works like that, I'd maybe move this check inside the first `if` statement below to help reading the function (as one unfamiliar with RegExp might be wondering if this is a bug to check to `/api/v2` on something we don't know yet if an endpoint and not a name)

**Review by lihbr — APPROVED** [review]
LGTM, great job!

File: src/client.ts (hunk 1 above)
Comment (angeloashmore) [reply src/client.ts]: Thanks, good catch

**Comment by angeloashmore** [comment]
Closed in favor of #224 (re-created PR after squashing #222)

**Pull request closed without merging**